.github/workflows/event-processor.yml

```diff
@@ -0,0 +1,155 @@
+name: GitHub Event Processor
+
+on:
+  issues:
+    types: [edited, labeled, opened, reopened, unlabeled]
+  # issue_comment is used for both issues and pull_requests
+  # github.event.issue.pull_request will be non-null on pull request comments
+  issue_comment:
+    types: [created]
+  # synchronize is the pull_request_target event when changes are pushed
+  # pull request merged is the closed event with github.event.pull_request.merged = true
+  pull_request_target:
+    types: [closed, labeled, opened, reopened, review_requested, synchronize, unlabeled]
+
+# This removes all unnecessary permissions, the ones needed will be set below.
+# https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token
+permissions: {}
+
+jobs:
+  # This event requires the Azure CLI to get the LABEL_SERVICE_API_KEY from the vault.
+  # Because the azure/login step adds time costly pre/post Az CLI commands to any every job
+  # it's used in, split this into its own job so only the event that needs the Az CLI pays
+  # the cost.
+  event-handler-with-azure:
+    permissions:
+      issues: write
+      pull-requests: write
+      # For OIDC auth
+      id-token: write
+      contents: read
+    name: Handle ${{ github.event_name }} ${{ github.event.action }} event with azure login
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'issues' && github.event.action == 'opened' }}
+    steps:
+      - name: 'Az CLI login'
+        uses: azure/login@v2
+        with:
+            client-id: ${{ secrets.AZURE_CLIENT_ID }}
+            tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+            subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+      - name: 'Run Azure CLI commands'
+        run: |
+          LABEL_SERVICE_API_KEY=$(az keyvault secret show \
+            --vault-name issue-labeler \
+            -n issue-labeler-func-key \
+            -o tsv \
+            --query value)
+
+          echo "::add-mask::$LABEL_SERVICE_API_KEY"
+          echo "LABEL_SERVICE_API_KEY=$LABEL_SERVICE_API_KEY" >> $GITHUB_ENV
```

Comment:
We will need to sync the labels to this repo before some of these processes will work.

Reply:
The repositories.txt file I updated a long time ago. Or are you talking about a different sync?

Reply:
I just want to be sure the labels are in-sync

Reply:
@heaths and @weshaggard, every repo in repositories.txt has its repository labels in the repo/labels store which is used by both the linter and event processor. The other piece of this is the AI label service, which is invoked during the initial issue triage rule. I've already chatted with @jsquire this morning and azure-sdk-for-rust is setup in the AI label service.